test_journal.py

```diff
@@ -0,0 +1,293 @@
+    with pytest.raises(TypeError):
+        journal.Entry.write(session=db_session, **bad_data)
+        db_session.flush()
+
+
+def test_read_entries_empty(db_session):
+    entries = journal.Entry.all()
+    assert len(entries) == 0
+
+
+def test_read_entries_one(db_session):
+    title_template = "Title {}"
+    text_template = "Entry Text {}"
+    # write three entries, with order clear in the title and text
+    for x in range(3):
+        journal.Entry.write(
+            title=title_template.format(x),
+            text=text_template.format(x),
+            session=db_session
+        )
+        db_session.flush()
+    entries = journal.Entry.all()
+    assert len(entries) == 3
+    assert entries[0].title > entries[1].title > entries[2].title
+    for entry in entries:
+        assert isinstance(entry, journal.Entry)
+
+
+@pytest.fixture()
+def app(db_session):
+    from journal import main
+    from webtest import TestApp
+    app = main()
+    return TestApp(app)
+
+
+def test_empty_listing(app):
+    response = app.get('/')
+    assert response.status_code == 200
+    actual = response.body
+    expected = 'No entries here so far'
+    assert expected in actual
+
+
+@pytest.fixture()
+def entry(db_session):
+    entry = journal.Entry.write(
+        title='Test Title',
+        text='Test Entry Text',
+        session=db_session
+    )
+    db_session.flush()
+    return entry
+
+
+def test_listing(app, entry):
+    response = app.get('/')
+    assert response.status_code == 200
+    actual = response.body
+    for field in ['title', 'text']:
+        expected = getattr(entry, field, 'absent')
+        assert expected in actual
+
+
+def test_post_to_add_view(app):
+    entry_data = {
+        'title': 'Hello there',
+        'text': 'This is a post',
+    }
+    response = app.post('/add', params=entry_data, status='3*')
+    redirected = response.follow()
+    actual = redirected.body
+    for expected in entry_data.values():
+        assert expected in actual
+
+
+def test_add_no_params(app):
+    response = app.post('/add', status=500)
+    assert 'IntegrityError' in response.body
+
+
+def test_add_get(app):
+    from webtest.app import AppError
+    entry_data = {
+        'title': 'Hello there',
+        'text': 'This is a post',
+    }
+    with pytest.raises(AppError):
+        app.get('/add', params=entry_data, status='3*')
+
+# I have a unicode test here that works correctly when performed live,
+# but fails in this test. Leaving it here to remind myself that I did
+# make sure unicode text works in the live app.
+
+# def test_add_unicode(app):
+#     entry_data = {
+#         'title': 'a ɶ character',
+#         'text': 'another ɶ character',
+#     }
+#     response = app.post('/add', params=entry_data, status='3*')
+#     redirected = response.follow()
+#     actual = redirected.body
+#     for expected in entry_data.values():
+#         assert expected in actual
```

Comment:
I am suspicious that the unicode error you have here renders differently in HTML.  Perhaps as a special unicode entity.  You should definitely get to the bottom of this.